### PR TITLE
Commit code for #1524 - Adding Header AccessControlMaxAge.

### PR DESCRIPTION
**Pull request by wpoosanguansit**
Simple commit for AccessControlMaxAge that defaults to 5 seconds.

### CODE REVIEW

File: zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * Access-Control-Allow-Methods and Access-Control-Allow-Headers headers) can be
+ * cached.
+ *
+ * Maximum number of seconds the results can be cached, as an unsigned
+ * non-negative integer. Firefox caps this at 24 hours (86400 seconds). Chromium
+ * (prior to v76) caps at 10 minutes (600 seconds). Chromium (starting in v76)
+ * caps at 2 hours (7200 seconds). The default value is 5 seconds.
+ */
+sealed trait AccessControlMaxAge {
+  val seconds: String
```
Comment (Wosin) [inline zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala]: Wouldn't it make more sense to have this as `Long` or `Duration` ?

File: zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  def fromAccessControlMaxAge(accessControlMaxAge: AccessControlMaxAge): String = {
+    accessControlMaxAge.seconds
+  }
+
+  def toAccessControlMaxAge(seconds: String): AccessControlMaxAge = {
+    Try(seconds.toLong).fold(
+      _ => ValidAccessControlMaxAge(),
+      long => if (long > 0) ValidAccessControlMaxAge(Duration(long, SECONDS)) else ValidAccessControlMaxAge(),
```
Comment (Wosin) [inline zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala]: Should we have a type to represent incorrect header value (like negative numbers) ??

File: zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala (hunk 1 above)
Comment (wpoosanguansit) [reply zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala]: I will change it to Duration.

File: zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala (hunk 2 above)
Comment (wpoosanguansit) [reply zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala]: I did have that but I will have that in again as I think it align closely with the spec, which says it is default to 5 seconds.

**Review by jdegoes — APPROVED** [review]

File: zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  val seconds: Duration
+}
+
+object AccessControlMaxAge {
+
+  /**
+   * Valid AccessControlMaxAge with an unsigned non-negative negative for
+   * seconds
+   */
+  final case class ValidAccessControlMaxAge(private val duration: FiniteDuration = Duration(5, SECONDS))
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala]: This should not used Scala duration, only `java.time.Duration`.

File: zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      extends AccessControlMaxAge {
+    override val seconds: Duration = duration
+  }
+
+  case object InValidAccessControlMaxAge extends AccessControlMaxAge {
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala]:
```suggestion
  case object InvalidAccessControlMaxAge extends AccessControlMaxAge {
```

`Invalid` is one word.

**Review by jdegoes — CHANGES_REQUESTED** [review]
Just the one change then good to merge.

File: zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala (hunk 3 above)
Comment (wpoosanguansit) [reply zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala]: Updated.

File: zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala (hunk 4 above)
Comment (wpoosanguansit) [reply zio-http/src/main/scala/zio/http/model/headers/values/AccessControlMaxAge.scala]: Updated.

**Comment by jdegoes** [comment]
Fails due to [compilation error](https://github.com/zio/zio-http/actions/runs/3265979566/jobs/5369042470#step:8:316)

**Comment by wpoosanguansit** [comment]
Update the call toSeconds to getSeconds() as toSeconds is not found in java8.

**Review by jdegoes — APPROVED** [review]

**Pull request merged**